File: examples/java-maven-example/src/test/java/org/galaxio/performance/picatinny/PerformanceSupport.java
package org.galaxio.performance.picatinny;

import io.gatling.javaapi.core.Assertion;

import java.time.Duration;
import java.util.List;
import java.util.concurrent.TimeUnit;

import static io.gatling.javaapi.core.CoreDsl.global;

public final class PerformanceSupport {
    private PerformanceSupport() {
    }

    public static scala.concurrent.duration.FiniteDuration toScala(Duration duration) {
        return scala.concurrent.duration.FiniteDuration.apply(duration.toNanos(), TimeUnit.NANOSECONDS);
    }

    public static scala.collection.immutable.Seq<io.gatling.commons.stats.assertion.Assertion> noFailedRequests() {
        Assertion assertion = global().failedRequests().count().is(0L);
        return scala.jdk.javaapi.CollectionConverters.asScala(List.of(assertion.asScala())).toSeq();
    }
}

File: examples/kotlin-gradle-example/src/gatling/kotlin/org/galaxio/performance/picatinny/PerformanceSupport.kt
package org.galaxio.performance.picatinny

import io.gatling.commons.stats.assertion.Assertion
import io.gatling.javaapi.core.CoreDsl.global
import scala.collection.immutable.Seq
import scala.concurrent.duration.FiniteDuration
import scala.jdk.javaapi.CollectionConverters
import java.time.Duration
import java.util.concurrent.TimeUnit

object PerformanceSupport {
    fun toScala(duration: Duration): FiniteDuration =
        FiniteDuration.apply(duration.toNanos(), TimeUnit.NANOSECONDS)

    fun noFailedRequests(): Seq<Assertion> {
        val assertion = global().failedRequests().count().shouldBe(0L)
        return CollectionConverters.asScala(listOf(assertion.asScala())).toSeq()
    }
}
